_source/_posts/2021-01-27-okta-ansible.md

```diff
@@ -0,0 +1,157 @@
+---
+layout: blog_post
+title: "Tutorial: Ansible and Account Automation with Okta"
+author: frederico-hakamine
+by: internal-contributor
+communities: [devops]
+description: "How to integrate Okta and Ansible to manage accounts in servers while abstracting account management from your playbooks"
+tags: [devops]
+tweets:
+- "Learn how to integrate Okta and Ansible to manage server accounts and abstract account management from your playbooks"
+- "Remove account management from your Ansible playbooks. Here's how"
+image: blog/
+type: conversion
+---
+
+**Tip:** This tutorial is part of our series on how to integrate Okta with popular Infrastructure as a Code solution. If you're not into Ansible, check out our [Puppet](/blog/2021/01/22/okta-puppet) and [Terraform](/blog/2020/04/24/okta-terraform-automate-identity-and-infrastructure) tutorials.
+
+I love using Ansible to deploy and manage configuration at scale. However, like any other configuration management solution, Ansible works best when playbooks don't change often. This is easy to accomplish when you manage only server artifacts and binaries, but it can get tricky if your playbooks manage user accounts and credentials that will change quite often as people join, change roles, or leave your team.
+
+To overcome this, admins would typically look at distributing SSH keys or using some sort of 3rd party LDAP/AD for login.
+
+Distributing SSH Keys has major maintenance implications since it requires a lot of work anytime you need to rotate keys. If you use SSH keys for user logins, the recommendation is to rotate keys often – to reduce risks in case the keys are compromised – plus every time an admin leaves the company.
+
+Integrating your infra to an AD/LDAP server – usually using a PAM module – maybe even more challenging, since this type of server is usually monolith and hosted on-prem. Getting a robust integration here is a tall order. From site-to-site VPNs (making your AD available everywhere) to adding additional servers (keeping compatibility across a variety of server distros), this is the kind of pain no Ops Engineer would like to sign up for.
```

Comment:
s/maybe even more challenging/may be even more challenging/

Reply:
s/usually monolith/usually monolithic/

Reply:
fixed